packages/loader/container-loader/src/container.ts

```diff
@@ -1879,7 +1879,19 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
 
     // Please avoid calling it directly.
     // raiseContainerWarning() is the right flow for most cases
-    private logContainerError(warning: ContainerWarning) {
+    private logContainerWarning(warning: ContainerWarning & { originalError?: unknown }) {
+        // ContainerWarnings typically wrap whatever specific error was caught,
+        // And expose a different errorType to consumers of the warning event.
+        // Let's log the original error "as-is" here if present
+        if (warning.originalError !== undefined) {
```

Comment:
Would it be better for this event to happen at the moment of wrapping, such that this knowledge does not need leak out to this layer? I feel like we either need to push it to be responsibility of the wrapper, or make it generic logic of logger, as we can't assume that every place that needs to log an error needs to have that kind of IF condition to log wrapped error. I.e. consider this problem to be more generic then just applicable to throttling - how would we solve it?